### PR TITLE
feat(wallet-template): accounts

### PR DESCRIPTION
**Pull request by ryanleecode**
#1933

Open the wallet popup -> ui/assets/wallet-popup.html

Click on Accounts

image

Add an Account

image

Set a keyset name

image

Select networks

image

Select Submit

image

See the Accounts

image

### CODE REVIEW

File: projects/wallet-template/src/containers/Accounts/index.tsx
Comment (kratico) [inline projects/wallet-template/src/containers/Accounts/index.tsx:1]: Remember to add the accounts routes to the `WalletPopup` container.
There shouldn't be a another independent `accounts.html` page for this.

Eventually, previous substrate connect extension pages (networks and bootnodes) should be routes in the wallet popu too

File: projects/wallet-template/src/containers/WalletPopup/pages/Accounts.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    await rpc.client.clearKeysets()
+    navigate(0)
+  }
+
+  useEffect(() => {
+    ;(async () => {
+      const existingKeySets = await rpc.client.listKeysets()
+      setKeysets(existingKeySets)
+    })()
+  }, [])
```
Comment (kratico) [inline projects/wallet-template/src/containers/WalletPopup/pages/Accounts.tsx:33]: use `useSwr` with https://swr.vercel.app/docs/mutation#bound-mutate

File: projects/wallet-template/src/containers/WalletPopup/pages/Accounts.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    </button>
+  )
+
+  type AccountItemProps = {
+    bgColor: string
+    text: string
+    subText: string
+  }
+
+  const AccountItem: React.FC<AccountItemProps> = ({
```
Comment (kratico) [inline projects/wallet-template/src/containers/WalletPopup/pages/Accounts.tsx:54]: `AccountItem`, `EmptyAccounts`, `Accounts` should be moved outside this `Accounts` component so they don't get redefined on every re-render

File: projects/wallet-template/src/containers/WalletPopup/pages/AddAccount/networks.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+export type Network = {
+  label: string
+  value: string
+  logo: string
+}
```
Comment (kratico) [inline projects/wallet-template/src/containers/WalletPopup/pages/AddAccount/networks.ts:5]: add `chainId` for each network

**Review by kratico — CHANGES_REQUESTED** [review]

File: projects/wallet-template/src/containers/WalletPopup/pages/Accounts.tsx (hunk 1 above)
Comment (ryanleecode) [reply projects/wallet-template/src/containers/WalletPopup/pages/Accounts.tsx:33]:
```
  const { data: keysets } = useSWR("/rpc/keysets", async () => {
    return rpc.client.listKeysets()
  })

```

just this works right?

Comment (kratico) [reply projects/wallet-template/src/containers/WalletPopup/pages/Accounts.tsx]: yes

for a naming convention, I would suggest `rpc.listKeysets`

`rpc.methodName`

**Review by kratico — APPROVED** [review]

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/parity-tech-update-for-march/7226/1

**Pull request merged**